### DISCUSSION TITLE
2d-3d image registration?

### DISCUSSION THREAD

**Discussion by mdhe1248** [SuperElastix/elastix#843] [Q&A]
I tried [2d-3d registration](https://github.com/SuperElastix/ElastixModelZoo/tree/master/models/Par0013) and have quite a few naive questions:

First, while the 2d fixed image is "front view", the registered image is "side view". Is it an expected result? Shouldn't it be the same orientation as the fixed image, or am I missing something here? It seems that the orientation of moving volumetric image is different from that of fixed. If I rotate the moving image to match the orientation to the fixed, the results seems reasonable (to me). 
(fixed 2d image)
![image](https://user-images.githubusercontent.com/13360917/163281955-5de834ab-17ab-465f-aafa-fdc5f3bd1f68.png)

(2d-3d registration)
![image](https://user-images.githubusercontent.com/13360917/163281197-dd899b9f-ab9f-4c5b-85ec-1e2050ad206c.png) 

(2d-3d registration after orientation change)
![image](https://user-images.githubusercontent.com/13360917/163281622-7568023f-03da-4aee-a014-665e40f55022.png)
 By the way, all images are upside-down; but this is not important here. This is because, my input images are upside-down.

Second, if I look at the [mhd header file](https://github.com/SuperElastix/ElastixModelZoo/blob/f6bc11f690062924a855474c0f13d089e6ad3666/models/Par0013/exampleData/xrayImage2.mhd#L7) of the fixed image (and also 3d image), it has `offset`. Where are they coming from? If I change them in the fixed image header, the resulting registered image is off-centered. 

Also, the values in `offset` are similar to those in `origin` in the [parameter file](https://github.com/SuperElastix/ElastixModelZoo/blob/f6bc11f690062924a855474c0f13d089e6ad3666/models/Par0013/Par0013Powell_NGC_singleImage.txt#L51). Are they related? It is hard to find why and how they are set. Please advise me. Along with this, [interpolator values](https://github.com/SuperElastix/ElastixModelZoo/blob/f6bc11f690062924a855474c0f13d089e6ad3666/models/Par0013/Par0013Powell_NGC_singleImage.txt#L53-L57) in the parameter file seem pretty important. If I don't set them (probably using default values), I get weird results. I don't understand how they are set. 
![image](https://user-images.githubusercontent.com/13360917/163277088-3fef7e67-fdb0-4f5d-8d0f-72d260b5f7a8.png)

Maybe related to my second question: Can I use an nrrd file as a fixed image? I loaded the example mhd file and save it as nrrd; and using it as a fixed image resulted in an off-centered, weird image:
![image](https://user-images.githubusercontent.com/13360917/163284295-058378f6-c7b6-4243-a3d5-019360ab09da.png)

My end goal is to register a 3d nrrd image to a 2d nrrd image. It would be nice if the 2d-3d registration demonstrated in this repository is applicable to many other data set.

**Comment by mdhe1248** [comment]
Progress update:
Well, the issue with NRRD format was the header. Saving as `.nrrd` did not keep all the header information in the `.mhd header file`. I ended up with writing `.nhdr` (NRRD header file) with all information in the mhd file.

Still, I don't understand the [interpolation parameters](https://github.com/SuperElastix/ElastixModelZoo/blob/f6bc11f690062924a855474c0f13d089e6ad3666/models/Par0013/Par0013Powell_NGC_singleImage.txt#L53-L57). Any comments?

**Comment by N-Dekker** [comment]
@ntatsisk Please check, as we discussed.

**Comment by ntatsisk** [comment]
Hi @mdhe1248, 

were you able to make it eventually work? I agree that a generalizable example of 2D-3D registration is important since it is a question that arises from time to time in the registration community.

I will also try to work in such an example in the coming month or so. Let me know if you made any progress with it.

**Reply by mdhe1248** [reply]
No, I couldn't make it.
I hope to hear from you or anyone else about 2d-3d registration in the near future.
